Source/Engine/Engine/RandomUtil.cs

```diff
@@ -24,5 +25,79 @@ public static float Rand()
         {
             return Random.Next(0, int.MaxValue) / (float)int.MaxValue;
         }
+
+        /// <summary>
+        ///Generates a random number within a specified range.
+        /// </summary>
+        /// <param name="min">The minimum value of the range.</param>
+        /// <param name="max">The maximum value of the range.</param>
+        /// <returns>The random <see cref="float"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static float RandRange(float min, float max)
+        {
+            return Random.NextFloat(min, max);
+        }
+
+        /// <summary>
+        ///Generates a random number within a specified range.
+        /// </summary>
+        /// <param name="min">The minimum value of the range.</param>
+        /// <param name="max">The maximum value of the range.</param>
+        /// <returns>The random <see cref="double"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static double RandRange(double min, double max)
+        {
+            return Random.NextDouble(min, max);
+        }
+
+        /// <summary>
+        /// Generates a random number within a specified range.
+        /// </summary>
+        /// <param name="min">The minimum value of the range.</param>
+        /// <param name="max">The maximum value of the range.</param>
+        /// <returns>The random <see cref="int"/>.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int RandRange(int min, int max)
+        {
+            return Random.Next(min, max);
+        }
+
+        /// <summary>
+        /// Generates a random point inside a sphere with radius 1.0.
+        /// </summary>
+        /// <returns>A random <see cref="Float3"/> within the sphere.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Float3 GetPointInsideSphere()
+        {
+            return Random.NextUnitVector3();
```

Comment:
I'm so sorry, I mixed up the "inside sphere" and "on sphere" cases.

On sphere: Random.NextUnitVector3();
Inside sphere: Flax is missing that method. I'd propose adding one to https://github.com/FlaxEngine/FlaxEngine/blob/master/Source/Engine/Utilities/Extensions.cs

The method could look like this. (Generalises to other dimensions as well)
I don't immediately have a good name for it though.
```csharp
Vector3 output;
            Real l;
            do
            {
                // Create random float with a mean of 0 and deviation of ±1
                output.X = NextFloat(random) * 2.0f - 1.0f;
                output.Y = NextFloat(random) * 2.0f - 1.0f;
                output.Z = NextFloat(random) * 2.0f - 1.0f;

                l = output.LengthSquared;
            } while (l > 1);
            return output;
```